dlc/src/main/scala/org/bitcoins/dlc/DLCMessage.scala

```diff
@@ -2,37 +2,502 @@ package org.bitcoins.dlc
+
+                (Sha256DigestBE(obj.values(sha256Index).str),
+                 Satoshis(obj.values(satsIndex).num.toLong))
+              }
+          }
+          .get
+          .toMap
+
+      val fundingInputs =
+        vec
+          .find(_._1 == "fundingInputs")
+          .map {
+            case (_, value) =>
+              value.arr.map { subVal =>
+                val obj = subVal.obj
+                val outpointIndex =
+                  obj.value.keys.toList.indexOf("outpoint")
+                val outputIndex = obj.value.keys.toList.indexOf("output")
+
+                OutputReference(
+                  TransactionOutPoint(obj.values(outpointIndex).str),
+                  TransactionOutput(obj.values(outputIndex).str))
+              }
+          }
+          .get
+          .toVector
+
+      val oracleInfo =
+        vec.find(_._1 == "oracleInfo").map(obj => OracleInfo(obj._2.str)).get
+
+      val pubKeys =
+        vec
+          .find(_._1 == "pubKeys")
+          .map {
+            case (_, value) =>
+              val obj = value.obj
+              val fundingKeyIndex =
+                obj.keys.toList.indexOf("fundingKey")
+              val toLocalCETKeyIndex =
+                obj.keys.toList.indexOf("toLocalCETKey")
+              val toRemoteCETKeyIndex =
+                obj.keys.toList.indexOf("toRemoteCETKey")
+              val finalAddressIndex =
+                obj.keys.toList.indexOf("finalAddress")
+
+              DLCPublicKeys(
+                ECPublicKey(obj.values(fundingKeyIndex).str),
+                ECPublicKey(obj.values(toLocalCETKeyIndex).str),
+                ECPublicKey(obj.values(toRemoteCETKeyIndex).str),
+                BitcoinAddress(obj.values(finalAddressIndex).str).get
+              )
+          }
+          .get
+
+      val totalCollateral = vec
+        .find(_._1 == "totalCollateral")
+        .map(obj => Satoshis(obj._2.num.toLong))
+        .get
+      val changeAddress =
+        vec
+          .find(_._1 == "changeAddress")
+          .map(obj => Bech32Address.fromString(obj._2.str).get)
+          .get
+      val feeRate =
+        vec
+          .find(_._1 == "feeRate")
+          .map(obj => SatoshisPerVirtualByte(Satoshis(obj._2.num.toLong)))
+          .get
+
+      val timeouts =
+        vec
+          .find(_._1 == "timeouts")
+          .map {
+            case (_, value) =>
+              val obj = value.obj
+              val penaltyIndex =
+                obj.keys.toList.indexOf("penalty")
+              val contractMaturityIndex =
+                obj.keys.toList.indexOf("contractMaturity")
+              val contractTimeoutIndex =
+                obj.keys.toList.indexOf("contractTimeout")
+
+              DLCTimeouts(
+                obj.values(penaltyIndex).num.toInt,
+                BlockTime(UInt32(obj.values(contractMaturityIndex).num.toLong)),
+                BlockTime(UInt32(obj.values(contractTimeoutIndex).num.toLong))
```

Comment:
this seems to be an extremely common pattern:
```
val index = obj.keys.toList.indexOf(keyName)
obj.values(index)
```
probably deserves a function or an implicit class to add a `get(keyName: String): Value` method